pkg/staticnetworkconfig/generator.go

```diff
@@ -0,0 +1,108 @@
+	FilePath     string
+	FileContents string
+}
+
+type StaticNetworkConfig interface {
+	GenerateStaticNetworkConfigData(hostsYAMLS string) ([]StaticNetworkConfigData, error)
+}
+
+type StaticNetworkConfigGenerator struct {
+	log logrus.FieldLogger
+}
+
+func New(log logrus.FieldLogger) StaticNetworkConfig {
+	return &StaticNetworkConfigGenerator{log: log}
+}
+
+func (s *StaticNetworkConfigGenerator) GenerateStaticNetworkConfigData(hostsYAMLS string) ([]StaticNetworkConfigData, error) {
+	hostsConfig := strings.Split(hostsYAMLS, StaticNetworkConfigHostDelimeter)
+	s.log.Infof("Start configuring static network for %d hosts", len(hostsConfig))
+	executer := &executer.CommonExecuter{}
+	filesList := []StaticNetworkConfigData{}
+	for _, hostConfig := range hostsConfig {
+		f, err := executer.TempFile("", "host-config")
+		if err != nil {
+			s.log.WithError(err).Errorf("Failed to create temp file")
+			return nil, err
+		}
+		_, err = f.WriteString(hostConfig)
+		if err != nil {
+			s.log.WithError(err).Errorf("Failed to write host config to temp file")
+			return nil, err
+		}
+		f.Close()
+		stdout, stderr, retCode := executer.Execute("nmstatectl", "gc", f.Name())
+		if retCode != 0 {
+			s.log.Errorf("nstatectl gc failed, errorCode %d, stderr %s, input yaml <%s>", retCode, stderr, hostConfig)
+			return nil, fmt.Errorf("nmstatectl failed, errorCode %d, stderr %s, input yaml <%s>", retCode, stderr, hostConfig)
+		}
+		err = s.createNMConnectionFiles(stdout, &filesList)
+		if err != nil {
+			s.log.WithError(err).Errorf("failed to create NM connection files")
+			return nil, err
+		}
+		os.Remove(f.Name())
+	}
+	return filesList, nil
+}
+
+func (s *StaticNetworkConfigGenerator) createNMConnectionFiles(nmstateOutput string, filesList *[]StaticNetworkConfigData) error {
+	var hostNMConnections map[string]interface{}
+	err := yaml.Unmarshal([]byte(nmstateOutput), &hostNMConnections)
+	if err != nil {
+		s.log.WithError(err).Errorf("Failed to unmarshal nmstate output")
+		return err
+	}
+	connectionsList := hostNMConnections["NetworkManager"].([]interface{})
+	for _, connection := range connectionsList {
+		connectionElems := connection.([]interface{})
+		fileName := connectionElems[0].(string)
+		fileContents, err := s.formatNMConnection(connectionElems[1].(string))
+		if err != nil {
+			return err
+		}
+		s.log.Infof("Adding NMConnection file <%s>", fileName)
+		newFile := StaticNetworkConfigData{
+			FilePath:     fileName,
+			FileContents: base64.StdEncoding.EncodeToString([]byte(fileContents)),
+		}
+		*filesList = append(*filesList, newFile)
+	}
+	return nil
+}
+
+// TODO - in case we will need additional formating, consider using https://github.com/go-ini/ini
+func (s *StaticNetworkConfigGenerator) formatNMConnection(nmConnection string) (string, error) {
+	start := strings.Index(nmConnection, "interface-name=")
+	if start == -1 {
+		s.log.Errorf("Failed to find interface-name label in nmconnection string")
```

Comment:
s/Errorf/Error (no need for formatting)

Reply:
i prefer to leave it with Errorf , since in case somebody will want to change it, he may not notice that this is Error and not Errorf, and i think there is no compilation error on it